admin/jobs/__init__.py

```diff
@@ -47,12 +48,177 @@ def create_files():
+            record = Application()
+            for idx, column in enumerate(columns):
+                if a[idx] != '':
+                    if column in ['status']:
+                        setattr(record, column, application_status(a[idx]))
+                    elif column in ['created', 'updated', 'downloaded', 'completed']:
+                        setattr(record, column, datetime.strptime(a[idx], '%Y-%m-%d  %H:%M:%S.%f'))
+                    else:
+                        val = str(f.decrypt(eval(a[idx])), 'utf-8')
+                        if val == 'True':
+                            val = True
+                        elif val == 'False':
+                            val = False
+                        elif val == 'None':
+                            val = None
+                        if val != None:
+                            setattr(record, column, val)
+
+            db.session.add(record)
+            db.session.commit()
+
+    except Exception as e:
+        print(e, flush=True)
+
+    return ('', 200)
+
+
+@jobs.route('/jobs/backup-files1', methods=['GET'])
+#@JobTokenRequired
+def backup_files1():
+    import io
+    import time
+    import zipfile
+    from grc.external_services.aws_s3_client import AwsS3Client
+
+    tic = time.perf_counter()
+
+    try:
+        awsclient = AwsS3Client()
+        awsclient_external = AwsS3Client(external=True)
+        zip_buffer = io.BytesIO()
+        all_files = awsclient.list_objects()
+
+        with zipfile.ZipFile(zip_buffer, 'x', zipfile.ZIP_DEFLATED, False) as zipper:
+            for object_name in all_files:
+                data = awsclient.download_object(object_name)
+                zipper.writestr(object_name, data.getvalue())
+
+        backup_file = f"{current_app.config['ENVIRONMENT'].lower()}/backups/{datetime.now().strftime('%Y%m%d%H%M')}.zip"
+        awsclient_external.upload_fileobj(zip_buffer, backup_file)
+
+    except Exception as e:
+        print(e, flush=True)
+
+    toc = time.perf_counter()
+    print(f"Finished in {toc - tic:0.4f} seconds", flush=True)
+
+    return ('', 200)
+
+
+@jobs.route('/jobs/backup-files2', methods=['GET'])
+#@JobTokenRequired
+def backup_files2():
+    import time
+    from zipstream import ZipStream
+    import pyAesCrypt
+    from grc.external_services.aws_s3_client import AwsS3Client
+
+    tic = time.perf_counter()
+
+    try:
+        awsclient = AwsS3Client()
+        awsclient_external = AwsS3Client(external=True)
+        all_files = awsclient.list_objects()
+        files = []
+
+        for key in all_files:
+            files.append({'stream': awsclient.stream_download_object(key), 'name': key})
+
+        zip_buffer = ZipStream(files, chunksize=32768)
+
+        fin = io.BytesIO()
+        for data in zip_buffer.stream():
+            fin.write(data)
+        fin.seek(0)
+
+        buffer_size = 64 * 1024
+        secret_key = os.environ.get('SQLALCHEMY_KEY', '')
```

Comment:
It would be great if we could use the same (new) encryption key for the external backups as we do for the database backups.
The GitHub secret is called `EXTERNAL_BACKUP_ENCRYPTION_KEY`
Maybe we could use the same name for a new environment variable

Reply:
Added, need the key to update CF